packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1154,6 +1158,16 @@ class SliverAppBar extends StatefulWidget {
   /// Defaults to [NavigationToolbar.kMiddleSpacing].
   final double titleSpacing;
 
+  /// The size of the app bar when it is collapsed.
+  ///
+  /// By default, the collapsed height is set to [kToolbarHeight]. If [bottom]
+  /// is set, its [bottom.preferredSize.height] is added to the height. If
+  /// [primary] is true, any top padding from [MediaQuery] is added as well.
```

Comment:
then the [MediaQuery] top padding, [MediauQueryData.padding.top], is added as well.